src/rdbms/mongoose_rdbms.erl

```diff
@@ -166,10 +168,29 @@
 -export_type([query_result/0,
               server/0]).
 
+-type options() :: #{driver := pgsql | mysql | odbc,
+                     max_start_interval := pos_integer(),
+                     atom() => any()}.
+
+-export_type([options/0]).
+
 %%%----------------------------------------------------------------------
 %%% API
 %%%----------------------------------------------------------------------
 
+-spec process_options(map()) -> options().
+process_options(Opts = #{driver := odbc, settings := _}) ->
+    Opts;
+process_options(Opts = #{host := _Host, database := _DB, username := _User,
+                         password := _Pass, driver := _Driver}) ->
+    ensure_db_port(Opts);
+process_options(Opts) ->
+    error(#{what => invalid_rdbms_connection_options, options => Opts}).
+
+ensure_db_port(Opts = #{port := _}) -> Opts;
```

Comment:
We cannot store ports as `defaults` because different DB engines use different ports, and it is not possible to set defaults for different outgoing pool types, right?

Reply:
Yes, this is correct.